parser/indexedfile.go

```diff
@@ -55,22 +57,35 @@ func newIndexedFile(filePath string, res *resources.Resources) (*fpt.IndexedFile
 	}
 	toReturn.SetHeader(header)
 
-	broDataFactory := pt.NewBroDataFactory(header.ObjType)
+	var broDataFactory func() pt.BroData
+	if header.ObjType != "" {
+		// TSV log files have the type in a header
+		broDataFactory = pt.NewBroDataFactory(header.ObjType)
+	} else if scanner.Err() == nil && len(scanner.Text()) > 0 && // no error and there is text
+		json.Valid(scanner.Bytes()) {
+		// JSON log files only have the type in the filename
+		broDataFactory = pt.NewBroDataFactory(filepath.Base(toReturn.Path))
+		toReturn.SetJSON()
+	}
```

Comment:
This is one of the main changes. Basically if the `scanHeader` call on line 53 above returns "" for the header type it *could* be because the log is in json. There are other reasons but that's why there is a check here to see if it is actually json. Next, json logs don't include any sort of header or indication on what type of log they are (e.g. conn, dns, http) so the only thing we have to go on is the filename.

Reply:
I can definitely rename the function. Is `scanTSVHeader` ok with you? The definition of scanHeader already has a note about advancing the file scanner, though I can edit it to match what you suggested. Do you want an additional note when it is called here?